src/PROPOSAL/PROPOSAL/Secondaries.h

```diff
@@ -51,11 +51,13 @@ class Secondaries {
 
     // Operational functions to fill and access track
     void reserve(size_t number_secondaries);
-    void clear() { track_.clear(); types_.clear(); };
-    void push_back(const ParticleState& point, const InteractionType& type);
+    void clear() { track_.clear(); types_.clear(); target_hashes_.clear(); };
+    void push_back(const ParticleState& point, const InteractionType& type,
+                   const size_t& target_hash = 0);
```

Comment:
Why does the hash get a default value?

Reply:
Mainly for backward compatibility for people who might have used the `push_back` function without specifying a target_hash